From 4eefdf0995c1ce440f51da22906e5f761a9a9c72 Mon Sep 17 00:00:00 2001
Message-Id: <4eefdf0995c1ce440f51da22906e5f761a9a9c72@dist-git>
From: Martin Kletzander <mkletzan@redhat.com>
Date: Fri, 4 Nov 2016 10:29:48 +0100
Subject: [PATCH] qemu: Disable migration with ivshmem

It was never safe anyway and as such shouldn't have been enabled in the
first place.  Future patches will allow hot-(un)pluging of some ivshmem
devices as a workaround.

Signed-off-by: Martin Kletzander <mkletzan@redhat.com>
(cherry picked from commit d17fab69be7a73336d388b805c282037ffb29647)

https://bugzilla.redhat.com/show_bug.cgi?id=1392031

Signed-off-by: Martin Kletzander <mkletzan@redhat.com>
---
 src/qemu/qemu_migration.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/qemu/qemu_migration.c b/src/qemu/qemu_migration.c
index 496a3f1..545ed1c 100644
--- a/src/qemu/qemu_migration.c
+++ b/src/qemu/qemu_migration.c
@@ -2350,6 +2350,12 @@ qemuMigrationIsAllowed(virQEMUDriverPtr driver,
                 return false;
             }
         }
+
+        if (vm->def->nshmems) {
+            virReportError(VIR_ERR_OPERATION_INVALID, "%s",
+                           _("migration with shmem device is not supported"));
+            return false;
+        }
     }
 
     return true;
-- 
2.10.2

